frappe/model/meta.py

```diff
@@ -358,17 +358,20 @@ def add_custom_fields(self):
 		if not frappe.db.table_exists("Custom Field"):
 			return
 
-		custom_fields = frappe.db.sql(
-			"""
-			SELECT * FROM `tabCustom Field`
-			WHERE dt = %s AND docstatus < 2
-		""",
-			(self.name,),
-			as_dict=1,
+		custom_fields = frappe.db.get_values(
+			"Custom Field",
+			filters={"dt": self.name},
+			fieldname="*",
+			as_dict=True,
+			order_by="idx",
 			update={"is_custom_field": 1},
 		)
 
+		if not custom_fields:
+			return
+
 		self.extend("fields", custom_fields)
+		self.flags.has_custom_fields = True
```

Comment:
Let not add one more flag for this? 

Meta is anyways cached we can just use `get_custom_fields` to see if there are any custom fields.